core/model-api/src/main/java/org/eclipse/rdf4j/model/Literal.java

```diff
@@ -280,7 +280,7 @@ default TemporalAmount temporalAmountValue() throws DateTimeException {
 	 *           match any of the core datatypes. Instead of minting a new Optional object every time this method is
 	 *           called it's recommended to use {@link CoreDatatype#asOptional()}.
 	 */
-	<T extends CoreDatatype> Optional<T> getCoreDatatype();
+	CoreDatatype getCoreDatatype();
```

Comment:
Happy with that - will need a javadoc tweak (but you've already got that on your TODO I saw 👍 )